pkg/systemlogmonitor/logwatchers/kmsg/log_watcher_linux.go

```diff
@@ -120,3 +124,35 @@ func (k *kernelLogWatcher) watchLoop() {
 		}
 	}
 }
+
+// create a new kmsg parser and sets it to the watcher.
+func (k *kernelLogWatcher) SetKmsgParser() error {
+	parser, err := kmsgparser.NewParser()
+	if err != nil {
+		return fmt.Errorf("failed to create kmsg parser: %v", err)
+	}
+	k.kmsgParser = parser
+	return nil
+}
+
+// revive ourselves if the kmsg channel is closed
```

Comment:
I am curious about the error you observe when the channel is closed. Do you see the "Kmsg channel closed" log? Is it always associated with high load of the node?